### PR TITLE
[RNTM-747] Collect python app files hierarchy

### PR DESCRIPTION
**Pull request by davidgonor1408**

### CODE REVIEW

File: lib/analyzer/applications/python/common.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+export function filterAppFiles(
+  filePaths: string[],
+): [string, ApplicationFileInfo[]] {
+  const appFiles: ApplicationFileInfo[] = [];
+  let rootDir: string = "";
+  const directories: Set<string> = new Set();
+
+  for (const filePath of filePaths) {
+    if (
+      !filePath.includes("site-packages/") &&
```
Comment (nozik) [inline lib/analyzer/applications/python/common.ts:13]: Why not "/site-packages/" / "/dist-packages/"?

File: lib/analyzer/applications/python/common.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+      !filePath.includes("dist-packages/") &&
+      // installed eternal packages are installed in `usr/` dir and should not be collected as app files
```
Comment (nozik) [inline lib/analyzer/applications/python/common.ts:15]: I would change the comment to "/usr/ should not include 1st party code"

File: lib/analyzer/applications/python/common.ts [hunk 3]
```diff
@@ -0,0 +1,36 @@
+import * as path from "path";
+import { ApplicationFileInfo } from "../types";
+
+export function filterAppFiles(
```
Comment (nozik) [inline lib/analyzer/applications/python/common.ts:4]: Is there a unit test that covers this logic?

File: lib/analyzer/applications/python/poetry.ts [hunk 4]
```diff
@@ -56,7 +88,7 @@ function findManifestLockPairsInSameDirectory(
 
   for (const directoryPath of Object.keys(fileNamesGroupedByDirectory)) {
     const filesInDirectory = fileNamesGroupedByDirectory[directoryPath];
-    if (filesInDirectory.length !== 2) {
+    if (filesInDirectory.length < 2) {
```
Comment (nozik) [inline lib/analyzer/applications/python/poetry.ts:91]: Can you explain this change?

File: lib/analyzer/applications/python/common.ts [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+): [string, ApplicationFileInfo[]] {
+  const appFiles: ApplicationFileInfo[] = [];
+  let rootDir: string = "";
+  const directories: Set<string> = new Set();
+
+  for (const filePath of filePaths) {
+    if (
+      !filePath.includes("/site-packages/") &&
+      !filePath.includes("/dist-packages/") &&
+      // "/usr/"" should not include 1st party code
```
Comment (nozik) [inline lib/analyzer/applications/python/common.ts:15]: Please remove redundant `"`s from comment

File: lib/analyzer/static-analyzer.ts [hunk 6]
```diff
@@ -200,10 +204,11 @@ export async function analyze(
     );
     const poetryDependenciesScanResults = await poetryFilesToScannedProjects(
       getFileContent(extractedLayers, getPoetryAppFileContentAction.actionName),
+      collectApplicationFiles,
```
Comment (nozik) [inline lib/analyzer/static-analyzer.ts:207]: Can't we collect the application files separately from poetry/pip? That could simplify the code and reduce the number of changes

Comment (davidgonor1408) [reply lib/analyzer/static-analyzer.ts:216]: Done

File: lib/analyzer/applications/python/pip.ts [hunk 7]
```diff
@@ -117,11 +117,14 @@ class PythonDepGraphBuilder {
  */
 export async function pipFilesToScannedProjects(
   filePathToContent: FilePathToContent,
+  collectApplicationFiles: boolean,
```
Comment (nozik) [inline lib/analyzer/applications/python/pip.ts:120]: All the changes to this file can be reverted, right?

File: lib/analyzer/applications/python/poetry.ts (hunk 4 above)
Comment (nozik) [reply lib/analyzer/applications/python/poetry.ts:60]: Do we still need this change after the recent commit?

File: lib/inputs/python/static.ts [hunk 8]
```diff
@@ -3,18 +3,20 @@ import { basename } from "path";
 import { ExtractAction } from "../../extractor/types";
 import { streamToString } from "../../stream-utils";
 
-const poetryManifestFiles = ["pyproject.toml", "poetry.lock"];
-const pipManifestFiles = ["requirements.txt"];
+const poetryFilePatterns = ["pyproject.toml", "poetry.lock", /\.py$/];
```
Comment (nozik) [inline lib/inputs/python/static.ts:6]: Can these changes be reverted?

File: lib/analyzer/applications/python/common.ts [hunk 9]
```diff
@@ -0,0 +1,78 @@
...
+export function getPythonApplicationFiles(
+  filePathToContent: FilePathToContent,
+) {
+  const scanResults: AppDepsScanResultWithoutTarget[] = [];
+
+  const [appFilesRootDir, appFiles] = filterAppFiles(
+    Object.keys(filePathToContent),
+  );
+
+  if (appFiles.length !== 0) {
```
Comment (nozik) [inline lib/analyzer/applications/python/common.ts:18]: if (!appfiles.length)

**Pull request merged**